internal/cli/atlas/accesslists/create.go

```diff
@@ -80,26 +83,85 @@ func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
 	return []*atlas.ProjectIPAccessList{entry}
 }
 
+func IPAddress() (string, error) {
+	if publicIP := store.IPAddress(); publicIP != "" {
+		return publicIP, nil
+	}
+
+	return "", errors.New("not able to find your public IP address. Specify the public IP address for this command")
+}
+
+func (opts *CreateOpts) needsArg() bool {
+	// Unless currentIP flag is enabled and type is ip address, args are required.
+	return !(opts.entryType == ipAddress && opts.currentIP)
+}
+
+func (opts *CreateOpts) validateCurrentIPFlag(cmd *cobra.Command, args []string) func() error {
+	return func() error {
+		if !opts.needsArg() && len(args) > 0 {
+			return fmt.Errorf(
+				"please either provide [entry] or use %s to use current IP Address.\n\nUsage: %s",
+				flag.CurrentIP,
+				cmd.UseLine(),
+			)
+		}
+
+		if opts.needsArg() && len(args) == 0 {
+			return fmt.Errorf(
+				"%q with entry type %s requires at least %d argument received %d\n\nUsage:  %s",
+				cmd.CommandPath(),
+				opts.entryType,
+				1,
+				len(args),
+				cmd.UseLine(),
```

Comment:
if we are keeping this let's then correct this format
```suggestion
			return fmt.Errorf(
				"%q with entry type %s requires at least 1 argument, received %d\n\nUsage:  %s",
				cmd.CommandPath(),
				opts.entryType,
				len(args),
				cmd.UseLine(),
```